pkg/process/metadata/workloadmeta/extractor_test.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+const (
+	Pid1 = 1000
+	Pid2 = 1001
+	Pid3 = 1002
+)
+
+func testProc(pid int32, cmdline []string) *procutil.Process {
+	return &procutil.Process{
+		Pid:     pid,
+		Cmdline: cmdline,
+		Stats:   &procutil.Stats{CreateTime: time.Now().Unix()},
+	}
+}
+
+func TestExtractor(t *testing.T) {
+	extractor := NewWorkloadMetaExtractor()
+	mockGrpcListener := new(mockGrpcListener)
+	extractor.grpcListener = mockGrpcListener
+
+	var (
+		proc1 = testProc(Pid1, []string{"java", "mydatabase.jar"})
+		proc2 = testProc(Pid2, []string{"python", "myprogram.py"})
+		proc3 = testProc(Pid3, []string{"corrina", "--at-her-best"})
+	)
+
+	// Assert that we write all procs on first run
```

Comment:
Let's also assert the state of the `WorkloadMetaExtractor` cache after every call to `Extract`. We're currently only checking what's been sent to the `gRPC` listener and not the internal state of the extractor. Wdyt?

Reply:
Sounds good to me! Added